imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -0,0 +1,485 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from imap_processing import cdf_utils, imap_module_directory
+from imap_processing.swe import swe_cdf_attrs
+
+# ESA voltage and index in the final data table
+esa_voltage_row_index_dict = {
+    0.56: 0,
+    0.78: 1,
+    1.08: 2,
+    1.51: 3,
+    2.10: 4,
+    2.92: 5,
+    4.06: 6,
+    5.64: 7,
+    7.85: 8,
+    10.92: 9,
+    15.19: 10,
+    21.13: 11,
+    29.39: 12,
+    40.88: 13,
+    56.87: 14,
+    79.10: 15,
+    110.03: 16,
+    153.05: 17,
+    212.89: 18,
+    296.14: 19,
+    411.93: 20,
+    572.99: 21,
+    797.03: 22,
+    1108.66: 23,
+}
+
+
+def read_lookup_table(table_index_value: int):
+    """Read lookup table from file.
+
+    Parameters
+    ----------
+    table_index_value : int
+        ESA table index number
+    """
+    lookup_table_filepath = f"{imap_module_directory}/swe/l1b/swe_esa_lookup_table.csv"
+    lookup_table = pd.read_csv(
+        lookup_table_filepath,
+        index_col="e_step",
+    )
+
+    if table_index_value == 0:
+        return lookup_table.loc[lookup_table["table_index"] == 0]
+
+    if table_index_value == 1:
+        return lookup_table.loc[lookup_table["table_index"] == 1]
+
+    raise ValueError("Error: Invalid table index value")
+
+
+def deadtime_correction(counts: np.array, tsampl: int):
+    """Calculate deadtime correction.
+
+    Deadtime correction is a technique used in various fields, including
+    nuclear physics,radiation detection, and particle counting, to compensate
+    for the effects of the time period during which a detector is not able to
+    record new events or measurements after detecting a previous event.
+    This "deadtime" is essentially the time during which the detector is
+    recovering from the previous detection and is unable to detect new events.
+
+    In particle detectors, there is a finite time required for the detector to
+    reset or recover after detecting a particle. During this deadtime, any
+    subsequent particles that may have arrived go undetected. As a result,
+    the recorded count rate appears to be lower than the actual count rate.
+
+    Deadtime correction involves mathematically adjusting the measured count
+    rates to compensate for this deadtime effect. This correction is crucial
+    when dealing with high-intensity sources or particle fluxes, as the deadtime
+    can significantly affect the accuracy of the measurements.
+
+    Deadtime correction is important to ensure accurate measurements and data
+    analysis in fields where event detection rates are high and where every
+    detected event is critical for understanding physical processes.
+
+
+
+    Parameters
+    ----------
+    counts : np.array
+        counts data before deadtime corrections
+    tsampl : int
```

Comment:
This parameter name is a bit confusing to me. Is it supposed to represent "time sample"?. Perhaps call it `time_sample`? or `acq_duration`? or just `duration`?

Reply:
I like `acq_duration`